server/lib/src/migration_data/dl8/access.rs

```diff
@@ -0,0 +1,2637 @@
+#![allow(clippy::expect_used)]
+//! Constant Entries for the IDM
+
+use crate::constants::uuids::*;
+use crate::entry::EntryInitNew;
+use crate::prelude::*;
+use crate::value::Value;
+use kanidm_proto::internal::Filter as ProtoFilter;
+
+lazy_static! {
+    /// either recycled or tombstone
+    pub static ref FILTER_RECYCLED_OR_TOMBSTONE: ProtoFilter = ProtoFilter::Or(vec![
+        match_class_filter!(EntryClass::Recycled),
+        match_class_filter!(EntryClass::Tombstone),
+    ]);
+
+    /// not either recycled or tombstone
+    pub static ref FILTER_ANDNOT_TOMBSTONE_OR_RECYCLED: ProtoFilter =
+        ProtoFilter::AndNot(Box::new(FILTER_RECYCLED_OR_TOMBSTONE.clone()));
+
+    /// members of 000000001000 / idm_high_privilege
+    pub static ref FILTER_HP: ProtoFilter = ProtoFilter::Eq(
+        Attribute::MemberOf.to_string(),
+        UUID_IDM_HIGH_PRIVILEGE.to_string(),
+    );
+
+    /// OR ( HP, Recycled, Tombstone)
+    pub static ref FILTER_HP_OR_RECYCLED_OR_TOMBSTONE: ProtoFilter = ProtoFilter::Or(vec![
+        FILTER_HP.clone(),
+        match_class_filter!(EntryClass::Recycled),
+        match_class_filter!(EntryClass::Tombstone),
+    ]);
+
+    pub static ref FILTER_ANDNOT_HP_OR_RECYCLED_OR_TOMBSTONE: ProtoFilter =
+        ProtoFilter::AndNot(Box::new(FILTER_HP_OR_RECYCLED_OR_TOMBSTONE.clone()));
+
+    pub static ref DEFAULT_TARGET_SCOPE: ProtoFilter = ProtoFilter::And(Vec::with_capacity(0));
+
+}
+
+#[derive(Clone, Debug, Default)]
+/// Who will receive the privileges of this ACP.
+pub enum BuiltinAcpReceiver {
+    #[default]
+    None,
+    /// This functions as an "OR" condition, that membership of *at least one* of these UUIDs
+    /// is sufficient for you to receive the access control.
+    Group(Vec<Uuid>),
+    EntryManager,
+}
+
+#[derive(Clone, Debug, Default)]
+/// Objects that are affected by the rules of this ACP.
+pub enum BuiltinAcpTarget {
+    #[default]
+    None,
```

Comment:
Just wondering. Wouldn't we want a invalid target to be unrepresentative? Why define None

Reply:
Because we model this in the ACP code underneath, I don't remember why I added it like this, but there was probably a reason, most likely so we could have a safe fallback in a parse error of some kind such that the acp will never apply.

Reply:
I seem to recall some migration related reason for it too ....

Reply:
Ahh found it. It's so that if a filter in the access control has no matches then we map it to a none so that it can clearly indicate no one actually gains the access, and we can shortcut out.

Reply:
And that then allows optimisation/caching of resolved access profiles which only map to conditions that apply to someone/something.